### ISSUE TITLE
#ifgroup throws exception: Collection was modified; enumeration operation may not execute.

### ISSUE THREAD

**Issue by markcandelora** [WorkMaze/JUST.net#104]
Using the transform: 

```JSON
{
    "tags": [
        {
            "#ifgroup(#exists($.tags.tag1))": {
                "Name": "tag1",
                "id": "96b7f975-3a4d-4020-abae-f7d17876ab39",
            }
        }
    ]
}
```

results in exception:
```
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at System.ThrowHelper.ThrowInvalidOperationException_InvalidOperation_EnumFailedVersion()
   at System.Collections.Generic.List`1.Enumerator.MoveNext()
   at JUST.JsonTransformer.RecursiveEvaluate(JToken parentToken, String inputJson, JArray parentArray, JToken currentArrayToken, JUSTContext localContext)
   at JUST.JsonTransformer.RecursiveEvaluate(JToken parentToken, String inputJson, JArray parentArray, JToken currentArrayToken, JUSTContext localContext)
   at JUST.JsonTransformer.RecursiveEvaluate(JToken parentToken, String inputJson, JArray parentArray, JToken currentArrayToken, JUSTContext localContext)
   at JUST.JsonTransformer.Transform(JObject transformer, JToken input, JUSTContext localContext)
   at JUST.JsonTransformer.Transform(JArray transformerArray, JToken input, JUSTContext localContext)
   at JUST.JsonTransformer.Transform(String transformerJson, JToken input, JUSTContext localContext)
   at JUST.JsonTransformer.Transform(String transformerJson, String inputJson, JUSTContext localContext)
   at Transformer.TransformMessage(String messageString) in C:\source\Transform.cs:line 202
   at TransformTest.Transform_TransformMessage() in C:\source\Tests\TransformTest.cs:line 25
```

**Comment by Courela** [comment]
After #108 being approved, I'll make a new pull request solving this issue.
